ui/components-react/src/components-react/propertygrid/MergingPropertyDataProvider.ts

```diff
@@ -0,0 +1,155 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module PropertyGrid
+ */
+
+import { PropertyRecord } from "@itwin/appui-abstract";
+import type {
+  IPropertyDataProvider,
+  PropertyCategory,
+  PropertyData,
+} from "./PropertyDataProvider";
+import { PropertyDataChangeEvent } from "./PropertyDataProvider";
+
+/**
+ * Implementation of [IPropertyDataProvider]. This provider merges any number of [IPropertyDataProviders] into a single one.
+ * @public
+ */
+export class MergingPropertyDataProvider implements IPropertyDataProvider {
+  /**
+   * Property data change event that is raised when any of the source providers raise this event.
+   */
+  public onDataChanged = new PropertyDataChangeEvent();
+  private _sourceProviderExtendedDataKey = "source-provider";
```

Comment:
I'd move this outside of the class, name it `SOURCE_PROVIDER_KEY` and prefix the value with `${UiComponents.packageName}/`.

Reply:
Moved